From 1e1ea532c9115c74b4927ca6d173645322c6c03d Mon Sep 17 00:00:00 2001
From: Jackie264 <jackie.han@gmail.com>
Date: Thu, 27 Aug 2026 10:07:04 +0800
Subject: [PATCH] Fix passwall2 configuration by removing bad keys

Remove 'disable' settings for tcp/udp_no_redir_ports in passwall2 configuration if present.
---
 .../root/etc/uci-defaults/luci-app-passwall2           | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/luci-app-passwall2/root/etc/uci-defaults/luci-app-passwall2 b/luci-app-passwall2/root/etc/uci-defaults/luci-app-passwall2
index 2175a7918..515c4ca83 100644
--- a/luci-app-passwall2/root/etc/uci-defaults/luci-app-passwall2
+++ b/luci-app-passwall2/root/etc/uci-defaults/luci-app-passwall2
@@ -32,6 +32,16 @@ uci commit firewall
 
 [ ! -s "/etc/config/passwall2" ] && cp -f /usr/share/passwall2/0_default_config /etc/config/passwall2
 
+if [ -s "/etc/config/passwall2" ]; then
+	_patched=0
+	for _bad_key in $(uci -q show passwall2 | grep -E "\.(tcp|udp)_no_redir_ports='disable'" | cut -d '=' -f 1); do
+		uci -q set "$_bad_key="
+		_patched=1
+	done
+	
+	[ "$_patched" -eq 1 ] && uci commit passwall2
+fi
+
 chmod +x /usr/share/passwall2/*.sh
 
 rm -f /tmp/luci-indexcache /tmp/luci-indexcache.*
